effects widgets tooltip UI polishing

Be-ing opened the pull request:
Make a new subclass of WKnobComposed that shows the effect parameter description as the tooltip. Previously effect parameter descriptions were only shown when hovering the cursor over effect parameter names, which is not easily discoverable.

Be-ing committed:
add WEffectParameterKnobComposed widget

This is needed to show effect parameter tooltips for knobs. Previously
they were only shown when hovering the cursor over effect parameter
names, which is not very easy to discover.

Be-ing committed:
clarify translator tip

Be-ing committed:
use new EffectParameterKnobComposed widget in skins

Shade is not being updated because it still uses the old Knob
widget, not KnobComposed.

Be-ing committed:
use new EffectParameterKnobComposed widget for EQ knobs

Be-ing committed:
always show full effect parameter name in tooltips

Be-ing committed:
use \n not : to separate effect parameter names & descriptions

Be-ing committed:
WEffectPushButton: show effect parameter description for tooltip

daschuer commented on res/skins/Deere/effect_parameter_knob.xml:
```diff
@@ -19,16 +19,19 @@
+            <EffectRack><Variable name="EffectRack"/></EffectRack>
+            <EffectUnit><Variable name="EffectUnit"/></EffectUnit>
+            <Effect><Variable name="Effect"/></Effect>
+            <EffectParameter><Variable name="EffectParameter"/></EffectParameter>
```
This is a source of errors. I think the knob can parse this info from the connected CO

Be-ing replied:
IMO it is not a big deal because the skin XML only needs to be done right in one XML template. Also, it is consistent with the other effects widgets and shares the same code on the C++ side to get the EffectRack/EffectChainSlot/EffectSlot/EffectParameterSlot.

Be-ing replied:
Getting that information from the connected CO would require writing new string parsing functions. I'd rather keep reusing the ones that already exist.

daschuer commented:
This PR discovers an issue with the Bandpass Tooltips of the GraphicEQs are zero based. 
Could you fix this along with this PR?

daschuer commented on src/skin/legacyskinparser.cpp:
```diff
@@ -553,6 +554,8 @@ QList<QWidget*> LegacySkinParser::parseNode(const QDomElement& node) {
     } else if (nodeName == "EffectSelector") {
         result = wrapWidget(parseEffectSelector(node));
+    } else if (nodeName == "EffectParameterKnobComposed") {
+        result = wrapWidget(parseEffectParameterKnobComposed(node));
```
Do you have a clue why the knob is called "composed"? How about ditching this suffix?

Be-ing replied:
I haven't looked at it in detail but KnobComposed does appear to be different from the old Knob widget used by Shade.

Be-ing replied:
The old Knob widget uses a series of images whereas the new KnobComposed widget uses a stationary background image and a single foreground image that is rotated.

daschuer commented on src/widget/weffectparameterknobcomposed.cpp:
```diff
@@ -0,0 +1,43 @@
+                       m_pEffectParameterSlot->name(),
+                       m_pEffectParameterSlot->description()));
+    } else {
+        setBaseTooltip(tr("No effect loaded."));
```
I think no one sees this, right? So there is no need to translate the string. How about using an empty sting?

daschuer commented on src/widget/weffectpushbutton.cpp:
```diff
@@ -86,6 +86,15 @@ void WEffectPushButton::mouseReleaseEvent(QMouseEvent* e) {
+                       m_pEffectParameterSlot->name(),
+                       m_pEffectParameterSlot->description()));
+    } else {
+        setBaseTooltip(tr("No effect loaded."));
```
The same here. Can we use an empty string?

daschuer commented:
Thank you for this nice PR. I have just tested it and it works already nice!

Be-ing committed:
don't have translators translate strings that won't be shown

Be-ing committed:
use 1-indexed numbers for graphic EQ parameter tooltips

Be-ing commented:
> This PR discovers an issue with the Bandpass Tooltips of the GraphicEQs are zero based.
Could you fix this along with this PR?

Why are the lowest and highest parameter knobs treated specially for their tooltips? I think it would make more sense to have them all numbered.

daschuer commented:
> Why are the lowest and highest parameter knobs treated specially for their tooltips? I think it would make more sense to have them all numbered.

Yes, or remove the numbers at all.

daschuer commented on src/skin/legacyskinparser.cpp:
```diff
@@ -1622,6 +1625,18 @@ QWidget* LegacySkinParser::parseEffectSelector(const QDomElement& node) {
     return pSelector;
 }
 
+QWidget* LegacySkinParser::parseEffectParameterKnobComposed(const QDomElement& node) {
```
I think we can use parseStandardWidget?

Be-ing replied:
No, the WEffectParameterKnobComposed constructor needs the EffectsManager* argument.

daschuer replied:
Ahh, OK.

daschuer commented:
I still do not like the redundant setup info that can go wrong. Parsing the key of the connected control is not too hard and keeps the skin xml clean: 

      if (!m_connections.isEmpty()) (
           ControlParameterWidgetConnection* connection = m_connections.at(0);
           const ConfigKey& configKey = connection->getKey();
           split configKey.group by "_" ....

Be-ing commented:
I do not like the idea of adding more string mangling. What if we decide to change the names of the ControlObjects in the future? That would break silently.

daschuer commented:
We can add a debug assert for that. 
It is better to add here a few lines of code here that will speed up skin loading and is less error prone, than adding redundant infos to the skin to reuse existing functions.

Be-ing committed:
get EffectParameterSlot from CO connected to knob

instead of making skins specify the required info in addition to
the connected parameter CO

Be-ing commented on src/widget/weffectparameterknobcomposed.cpp:
```diff
@@ -1,37 +1,54 @@
+const QString groupClose = "]";
+} // anonymous namespace
+
+void WEffectParameterKnobComposed::setupEffectParameterSlot(const ConfigKey& configKey) {
```
IMO this function is quite hacky, but if you prefer doing it this way I don't care much either way.

Be-ing commented:
When this is merged I will work on refactoring Deere to remove the EQ knob labels and the bottom splitter that confuses users and serves no purpose other than hiding the EQ knob labels. I'll also add support for 4 knob EQs to the skins.

daschuer commented:
Thank you very much for this and the latest changes. LGTM
